pkg/gui/password.go

```diff
@@ -99,8 +107,10 @@ func PasswordDeleteDialog(nickname string, app *fyne.App) chan string {
 }
 
 // This function is blocking, it returns when the user submit or cancel the form.
-func Password(nickname string, app *fyne.App) string {
-	return <-PasswordDialog(nickname, app)
+func Password(nickname string, app *fyne.App) PasswordEntry {
```

Comment:
Your password function is exposing its internal logic: You should return a string and an error.

As explained above the code consumming this function feel weird because of this.

My suggestion: 
```suggestion
func Password(nickname string, app *fyne.App) (string, error) {
```